apps/veil/src/pages/tournament/ui/previous-epochs.tsx

```diff
@@ -11,35 +12,120 @@ import { Density } from '@penumbra-zone/ui/Density';
 import { Button } from '@penumbra-zone/ui/Button';
 import { Text } from '@penumbra-zone/ui/Text';
 import { connectionStore } from '@/shared/model/connection';
-import { usePreviousEpochs, BASE_PAGE, BASE_LIMIT, EpochVote } from '../api/use-previous-epochs';
+import { usePreviousEpochs, BASE_PAGE, BASE_LIMIT } from '../api/use-previous-epochs';
+import type { PreviousEpochData } from '../server/previous-epochs';
 import { useSortableTableHeaders } from './sortable-table-header';
+import { usePersonalRewards } from '../api/use-personal-rewards';
+import { GENERIC_DELUM } from './shared/delum-metadata';
 import { Vote } from './vote';
 
 const TABLE_CLASSES = {
   table: {
     default: cn('grid-cols-[200px_1fr_48px]'),
-    connected: cn('grid-cols-[200px_1fr_160px_160px_48px]'),
+    connected: cn('grid-cols-[200px_1fr_200px_48px]'),
   },
   row: {
     default: cn('col-span-3'),
-    connected: cn('col-span-5'),
+    connected: cn('col-span-4'),
   },
 };
 
+interface PreviousEpochsRowProps {
+  row: PreviousEpochData;
+  isLoading: boolean;
+  className: string;
+  connected: boolean;
+}
+
+const PreviousEpochsRow = ({ row, isLoading, className, connected }: PreviousEpochsRowProps) => {
+  const { data: rewards, isLoading: rewardsLoading } = usePersonalRewards(row.epoch);
+
+  const votingRewards = useMemo(() => {
+    const reward = rewards?.[0];
```

Comment:
Not sure if this will work once we have different rewards for different votes based on the delegation asset.